### DISCUSSION TITLE
Weakness in the Generic SANS Calculator tool

### DISCUSSION THREAD

**Discussion by smk78** [SasView/sasview#2351] [General]
Biological User PeterH comments (the following is from several emails):

>I believe that the [Generic SANS Calculator] tool isn’t viable for use with soft matter systems, such as proteins, due to the definition used for the natural volume of an atom.
>
>On the [website](https://www.sasview.org/docs/user/qtgui/Calculators/sas_calculator_help.html) it mentions that _“V can be corrected by users (input parameter Total volume). This correction is useful especially for an atomic structure (such as taken from a PDB file) to get the right normalization”_ but I don’t believe this is the correct way to normalise the profile. 
>
>I have looked at the source code and the calculation depends on the SLD of an atoms defined by the .pdb file you upload.....The weighting of the scattering contribution from each atom is `(SLD_atom-SLD_solvent)*Volume_of_atom`. It is worth noting that when `SLD_solvent = 0` then `SLD_atom*SLD_solvent = the correct scattering length of the atom`. 
>
>The problem is that the SLD of the atoms are defined by volumes that wouldn’t be applicable in a protein and the correct SL of the atom i.e. `SLD_atom =  SL_atom/Vol_atom`. This results (for soft matter systems) in incorrect weighing of the scattering contributions.
>
>My proposed solution would be to recalculate an SLD for each atom that may be found in a protein (C,N,O,H,D,S etc.) and calculate am empirical SLD for an atom (instead of one based on “normal materials”) so that SLD of the entire protein would be the same for an atomic treatment as it would for calculating the SLD of the entire protein based on the SLDs of amino acids [as, for example, happens in the calculator in [SASSIE-Web](https://sassie-web.chem.utk.edu/sassie2/) or the [Biological Scattering Calculator](http://psldc.isis.rl.ac.uk/) (also see SasView/sasmeta#3 )].

**Comment by smk78** [comment]
If we agree with Peter, then at the very least a warning in the docs that the calculator may not be appropriate for biological and some other soft matter systems might be in order?

Kind of related to this is this piece of work:
[https://doi.org/10.1088/1361-648X/abfc13](https://doi.org/10.1088/1361-648X/abfc13)
"Thermal neutron cross sections of amino acids from average contributions of functional groups"
Romanelli et al, J. Phys. Cond. Mat., 2021

**Comment by butlerpd** [comment]
Hummm... I need to think a bit about this but I am a bit confused at the moment of why we would use SLD in this case. SLD should only be used for groups of atoms I would think? such as amino acids or, as we used to do a lot in the old days, CH3 groups, CH2 groups etc where indeed we used tabulated values for those groups in similar materials.

But, given we have a real space "atomic" structure from the PDB don't we have all the necessary information to do the calculation properly (modula solvent effects perhaps)? we have the atom types and their coordinates? Why would we be guessing at an effective atomic SLD? Of course I may not understand the PDB format (ok .. I know nothing really 😄 so maybe we don't have the coordinates?

**Comment by butlerpd** [comment]
Looking at this again, I think the user was confused by our poor nomenclature. I think this based on reviews I did of the code for the beta and Rg calculations added by @smalex-z to the generic scattering calculator where terms were labeled sld even though they were clearly b (in fm). the bs are available from `periodictable` so I'm pretty sure that is what is happening. the problem is the volume calculated for the object is I think based on those atomic volumes which are incorrect and lead to a volume that can be a factor of 2 off.
